msteams-platform/concepts/build-and-test/deep-link-application.md

```diff
@@ -0,0 +1,259 @@
+
+# [TeamsJS v2](#tab/teamsjs-v2)
+
+To implement, add a **copy link** action to each item, in whatever way best suits your UI. When the user takes this action, call [`pages.shareDeepLink()`](/javascript/api/@microsoft/teams-js/pages?view=msteams-client-js-latest#@microsoft-teams-js-pages-sharedeeplink&preserve-view=true) to display a dialog containing a link that the user can copy to the clipboard. When you make this call, pass an ID for your item. You get it back in [context](~/tabs/how-to/access-teams-context.md) when the link is followed and your tab is reloaded.
+
+```javascript
+pages.shareDeepLink({ subPageId: <subPageId>, subPageLabel: <subPageLabel>, subPageWebUrl: <subPageWebUrl> })
+```
+
+You'll need to replace the fields with the appropriate information:
+
+* `subPageId`: A unique identifier for the item within your page to which you're deep linking.
+* `subPageLabel`: A label for the item to use for displaying the deep link.
+* `subPageWebUrl`: A fallback URL to use if the client can't render the page.
+
+For more information, see [pages.shareDeepLink()](/javascript/api/@microsoft/teams-js/pages?view=msteams-client-js-latest#@microsoft-teams-js-pages-sharedeeplink&preserve-view=true).
+
+# [TeamsJS v1](#tab/teamsjs-v1)
+
+To implement this, add a **copy link** action to each item, in whatever way best suits your UI. When the user takes this action, call `shareDeepLink()` to display a dialog containing a link that the user can copy to the clipboard. When you make this call, pass an ID for your item. You get it back in [context](~/tabs/how-to/access-teams-context.md) when the link is followed and your tab is reloaded.
+
+```javascript
+microsoftTeams.shareDeepLink({ subEntityId: <subEntityId>, subEntityLabel: <subEntityLabel>, subEntityWebUrl: <subEntityWebUrl> })
+```
+
+You'll need to replace the fields with the appropriate information:
+
+* `subEntityId`: A unique identifier for the item within your tab to which you're deep linking.
+* `subEntityLabel`: A label for the item to use for displaying the deep link.
+* `subEntityWebUrl`: An optional field with a fallback URL to use if the client doesn't support rendering the tab.
+
+---
+
+Alternatively, you can also generate deep links programmatically using the format specified later in this article. You can use deep links in [bot](~/bots/what-are-bots.md) and [connector](~/webhooks-and-connectors/what-are-webhooks-and-connectors.md) messages that inform users about changes to your tab, or to items within it.
+
+> [!NOTE]
+> This deep link is different from the links provided by the **Copy link to tab** menu item, which just generates a deep link that points to this tab.
+
+>[!IMPORTANT]
+> Currently, `shareDeepLink` doesn't work on mobile platforms.
+
+### Consume a deep link from a tab
+
+When navigating to a deep link, Microsoft Teams navigates to the tab and provides a mechanism through the TeamsJS library to retrieve the subpage ID if it exists.
+
```

Comment:
This should be covered in "handling" deep links section

Reply:
Moved it to handling deep links section